### PR TITLE
[Core] Split internal interface process minor enhancements

### PR DESCRIPTION
**Pull request by rubenzorrilla**
**Description**
This adds the following minor enhancements to the `SplitInternalInterfaceProcess`:
- Adding the option to save the interface nodes and conditions in a user-defined modelpart
- Creation of a new properties container for each interface (so far the interface settings were saved in the properties 1, something that I found quite counter intuitive)
- The Python process factory file has been renamed to match the cpp ones, something that might eventually ease going through the code.

### CODE REVIEW

File: kratos/processes/split_internal_interfaces_process.cpp [hunk 1]
```diff
@@ -72,24 +88,30 @@ void SplitInternalInterfacesProcess::SplitBoundary(const std::size_t PropertyIdB
     max_node_id++;
 
     std::map<std::size_t, Node<3>::Pointer> new_nodes_map;
```
Comment (philbucher) [inline kratos/processes/split_internal_interfaces_process.cpp:116]: why not unordered?

Comment (rubenzorrilla) [reply kratos/processes/split_internal_interfaces_process.cpp:116]: Not 100% sure as this was already there, but I think that for the proper construction of the interface geometries we need to keep the insertion order, which is done base in the original nodes one.

**Comment by rubenzorrilla** [comment]
Ping @maceligueta and @RiccardoRossi. Note that this is required for the proper development of the coupling condition solver.

**Comment by rubenzorrilla** [comment]
While trying the process I found a bug which is also fixed in here. Basically we were not updating the ids. in the non-neighbour elements. The fix basically consists in creating an auxiliary map with the condition parents (which I assume to be unique, meaning that internal conditions will not be considered), to then apply the ids. update also to these conditions.

**Comment by rubenzorrilla** [comment]
Ping @maceligueta @RiccardoRossi. Please note that these changes require to be merged before pushing the coupling thermal condition branch.

**Comment by maceligueta** [comment]
> While trying the process I found a bug which is also fixed in here. Basically we were not updating the ids. in the non-neighbour elements. The fix basically consists in creating an auxiliary map with the condition parents (which I assume to be unique, meaning that internal conditions will not be considered), to then apply the ids. update also to these conditions.

I am afraid I don't really understand what the bug was. The idea of the original implementation was to duplicate nodes, so that the old node keeps the id, while the copy gets a new id which is greater than the max id existing in the modelpart. The new conditions created as joint elements also obtained ids greater than the existing ones. Can you elaborate a bit more what was the problem plz @rubenzorrilla ?

**Comment by rubenzorrilla** [comment]
> > While trying the process I found a bug which is also fixed in here. Basically we were not updating the ids. in the non-neighbour elements. The fix basically consists in creating an auxiliary map with the condition parents (which I assume to be unique, meaning that internal conditions will not be considered), to then apply the ids. update also to these conditions.
> 
> I am afraid I don't really understand what the bug was. The idea of the original implementation was to duplicate nodes, so that the old node keeps the id, while the copy gets a new id which is greater than the max id existing in the modelpart. The new conditions created as joint elements also obtained ids greater than the existing ones. Can you elaborate a bit more what was the problem plz @rubenzorrilla ?

Aside of the minor usability issues, the bug was the following. So far we checked the interface to be split with the neighbors, and from this we updated the connectivities in the "new" side. However, this doesn't take care of the nodes belonging to an element that does not share a face with the newly created interface. In other words, we need to update the connectivities of all the elements in one side, not only those that have an interface coinciding face. The fix includes this.

**Comment by maceligueta** [comment]
Thanks Ruben.

**Review by maceligueta — APPROVED** [review]

**Pull request merged**